proto/streamlit/proto/NumberInput.proto

```diff
@@ -19,9 +19,27 @@ syntax = "proto3";
 message NumberInput {
   string id = 1;
   string label = 2;
-  float default = 3;
-  float min = 5;
-  float max = 6;
-  float step = 7;
-  string format = 8;
+
+  oneof data {
+    IntNumberInput int_data = 3;
+    FloatNumberInput float_data = 4;
+  }
+
+  string format = 5;
+  bool has_min = 6;
+  bool has_max = 7;
```

Comment:
We shouldn't reuse numbers for proto fields. So the proper way to do this would be:

```proto
message NumberInput {
  string id = 1;
  string label = 2;

  oneof data {
    IntNumberInput int_data = 9;
    FloatNumberInput float_data = 10;
  }

  string format = 8;
  bool has_min = 11;
  bool has_max = 12;

  // deprecated: 3, 5, 6, 7
}
```

Reply:
There's a `reserved` protobuf keyword we could use to enforce this: https://developers.google.com/protocol-buffers/docs/proto3#reserved